datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/metadata.py

```diff
@@ -166,6 +167,21 @@
 
 MAX_DESCRIPTION_LENGTH = 400
 
+METRIC_REPLACEMENT = re.compile(r"([^a-zA-Z0-9_.]+)|(^[^a-zA-Z]+)")
+METRIC_DOTUNDERSCORE_CLEANUP = re.compile(r"_*\._*")
+
+
+def normalize_metric_name(metric_name):
+    """Copy pasted from the backend normalization code.
+    Metrics in metadata.csv need to be formatted this way otherwise, syncing metadata will fail.
+    Function not exported as a util, as this is different than AgentCheck.normalize. This function just makes sure
+    that whatever is in the metadata.csv is understandable by the backend.
+    """
+    if not isinstance(metric_name, string_types):
+        metric_name = str(metric_name)
+    metric_name = METRIC_REPLACEMENT.sub("_", metric_name)
+    return METRIC_DOTUNDERSCORE_CLEANUP.sub(".", metric_name).strip("_")
```

Comment:
Should we reuse this:

https://github.com/DataDog/integrations-core/blob/1bf1bcf5f4af4a54f16bf45ad76d55c4f364ce05/datadog_checks_base/datadog_checks/base/checks/base.py#L601-L625

So we are certain that we are using the same normalisation rule?  

Probably need to extract `self.normalize` to an util function first.

Reply:
> I didn't extract the normalization as a generic utils method as we already have a normalization method inside the AgentCheck base class that works slightly differently. This one is really about making sure that the backend won't fail.

Reply:
I don't think we want to be certain that we're using the same normalization rule.

This AgentCheck.normalize method is doing things differently (like remapping unicode) where the backend will normalize unicode to an underscore.

I think it's best not to mix them up, the one I added is really about making sure the metadata.csv is valid.

Reply:
My bad. Thx for the quote.

Reply:
> So we are certain that we are using the same normalisation rule?

We are not :)